File: app/src/main/java/me/imzack/app/ender/injector/component/GuideComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.GuidePresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.GuideActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = GuidePresenterModule.class)
public interface GuideComponent {

    void inject(GuideActivity guideActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/GuideComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.GuidePresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.GuideActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(GuidePresenterModule::class))
interface GuideComponent {

    fun inject(guideActivity: GuideActivity)
}
